compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5961,6 +5961,14 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr, AnalyzerD
                         DiagnosticWarningCode.CHECKED_EXPR_INVALID_USAGE_NO_ERROR_TYPE_IN_RHS,
                         operatorType);
                 data.resultType = checkedExpr.expr.getBType();
+
+                BLangTypeConversionExpr impConversionExpr = checkedExpr.expr.impConversionExpr;
+                if (impConversionExpr != null) {
+                    // Exclude any manually added error type, by reassigning
+                    BType expType = Types.getReferredType(data.expType);
```

Comment:
Why did you have to take the actual type here?

Reply:
We do that when we set the `ImplicitCastExpr`. I'm sticking to the same behavior to avoid any inconsistency. 

https://github.com/ballerina-platform/ballerina-lang/blob/39d844eb7ee4f780584be4ad0d9f7322bb7e4def/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L2408-L2409

Reply:
I think it's okay to use the `data.expType` here instead of the actual type, since it's the expected type of the expression.

Reply:
Please check with the new changes. Needed if-check located inside `setImplicitCastExpr()` also. Without duplicating logic called `setImplicitCastExpr()` here.